server/src/services/user.ts

```diff
@@ -0,0 +1,17 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import User from '../database/models/user';
+
+/**
+ * Finds a user by its id.
+ * @param {number} userId - The id of the user to be found.
+ * @returns {Promise<User>} - A promise that resolves with the found user object.
+ * @throws {Error} - If the user is not found, it throws an error with a message indicating the missing user with the specific id.
+ */
+export async function findUserById(userId: number): Promise<User> {
+  const user: User | null = await User.findByPk(userId);
+  if (!user) throw new Error (`user with id ${userId} not found`);
```

Comment:
For instance, here: could we have a new class extending `Error` for this error? I think this is partially up to preference, and the present method could just as well be suitable in the TypeScript world

Reply:
Yes, we could! Don't know if we could utilize any of the [sequelize errors](https://sequelize.org/api/v6/identifiers.html#errors) here, or maybe these are more for the sequelize internals and not for the actual query results. Anyways this needs some careful thinking before implementing.